operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,27 +148,23 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		retrigErr = err
 	}
 
+	// Determine if the personal namespace should be deleted
+
+	nsName := fmt.Sprintf("tenant-%s", strings.ReplaceAll(tn.Name, ".", "-"))
+
+	// Test if namespace has been open for too long; attempt to delete if there are no instances inside
+	keepNsOpen, err := r.enforceNamespaceKeepAliveOrDelete(ctx, &tn, nsName, r.TenantWorkspaceKeepAlive)
```

Comment:
To me, the name of this function is misleading, since it only returns whether the tenant namespace should be present or not, while performing no operations on it.

Reply:
Resolved in [8d122ff](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/8d122ffeb4464c5a8e1ec6c85d7aace5e6162713)